storage/jellyfish_merkle/src/node_type/node_type_test.rs

```diff
@@ -81,6 +82,29 @@ fn test_encode_decode() {
     }
 }
 
+#[test]
+fn test_internal_validity() {
+    let result = panic::catch_unwind(|| {
```

Comment:
I agree with @wqfish  it's more straightforward to split the case into two and use `should_panic`